hangups/client.py

```diff
@@ -42,6 +41,8 @@ class Client(object):
             retry_backoff_base^(# of retries attempted thus far)
             Defaults to 2.
     """
+    # http_utils.ClientSession instance (populated by .connect()):
+    _session = None
```

Comment:
Why is this a class attribute now?

Reply:
To allow a graceful cleanup in case of a failed instance init.

Reply:
How does the cleanup happen in that case?

Reply:
`__del__` is called by the garbage collector before it is destructed. And without the class variable a `AttributeError` is raised.
here is a demo: https://gist.github.com/das7pad/965fe09f94e79fab03373445f801bf98